### PR TITLE
Feature/button icons color

### PR DESCRIPTION
**Pull request by eddiman**
Related to "added support for hiding new fields in button section, and added new image radio button in library" #15 in h5p-editor-course-presentation
  
  Adds support for changing settings when module is displayed as button

### CODE REVIEW

File: src/styles/cp.css [hunk 1]
```diff
@@ -427,7 +446,7 @@
 .h5p-interaktivtavle .h5p-element-button:before {
   text-decoration: none;
   font-family: 'H5PFontAwesome4';
-  content: "\f119";
+  /**content: "\f119";*/
```
Comment (kentis) [inline src/styles/cp.css:449]: fjern utkommentert kode, stol på scm

**Review by kentis — CHANGES_REQUESTED** [review]

File: semantics.json [hunk 2]
```diff
@@ -159,6 +159,88 @@
                       }
                     ]
                   },
+                  {
+                    "name": "buttonColor",
+                    "type": "text",
+                    "widget": "colorSelector",
+                    "label": "Button Color",
```
Comment (boyum) [inline semantics.json:166]: Do we need translations for this?

File: src/scripts/cp.js [hunk 3]
```diff
@@ -1118,7 +1125,51 @@ CoursePresentation.prototype.createInteractionButton = function (element, instan
 
   return $button;
 };
-
+/**
+ * Checks whether a color is defined as dark or light 
+* @param originalColor, hex color code
+ * @return string, 'light' or 'dark'
```
Comment (boyum) [inline src/scripts/cp.js:1131]:
```suggestion
 * @return {'light' | 'dark'}
```

File: src/scripts/cp.js [hunk 4]
```diff
@@ -1118,7 +1125,51 @@ CoursePresentation.prototype.createInteractionButton = function (element, instan
...
+
+  if (!originalColor.includes(hashtag)) { color = hashtag.concat(originalColor) }
+  // Check the format of the color, HEX or RGB?
+  if (color && color.match(/^rgb/)) {
+    // If HEX --> store the red, green, blue values in separate variables
+    color = color.match(/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/)
+
+    r = color[1];
+    g = color[2];
+    b = color[3];
```
Comment (boyum) [inline src/scripts/cp.js:1147]:
```suggestion
    [_, r, g, b] = color.match(/^rgba?\((\d+),\s*(\d+),\s*(\d+)(?:,\s*(\d+(?:\.\d+)?))?\)$/);
```

File: src/styles/cp.css [hunk 5]
```diff
@@ -414,7 +414,26 @@
 
 .h5p-interaktivtavle .h5p-element-button:hover {
   box-shadow: 0 0.3em 0.25em 0 rgba(0, 0, 0, 0.5);
-  background: #003fdf;
+ /**TODO: Remove this comment, only for remembering the hex/
+  /* background: #003fdf;*/
```
Comment (boyum) [inline src/styles/cp.css:418]:
```suggestion
```

File: src/styles/cp.css (hunk 1 above)
Comment (boyum) [inline src/styles/cp.css:449]:
```suggestion
```

File: semantics.json (hunk 2 above)
Comment (eddiman) [reply semantics.json:166]: I can add the fields for them in the language json, but I dont see that the surrounding fields (buttonSize, displayAsButton etc,) gets translated either, when a different language in localhost is applied

**Review by kentis — APPROVED** [review]

File: .github/workflows/build_and_test.yml [hunk 6]
```diff
@@ -26,23 +26,20 @@ jobs:
...
-        if [ -d "h5p-open-ended-question" ]; then pushd h5p-open-ended-question && npm install && npm run build; popd; fi
-        if [ -d "h5p-simple-multiple-choice" ]; then pushd h5p-simple-multiple-choice && npm install && npm run build; popd; fi
+        if [ -d "h5p-editor-color-selector" ]; then cd h5p-editor-color-selector && git checkout b0ae8c941cedcec73df6b186663eae6df8679810; fi
+        if [ -d "h5p-dialogcards" ]; then cd h5p-dialogcards && npm install && npm run build; fi
+        if [ -d "h5p-drag-text" ]; then cd h5p-drag-text && npm install && npm run build; fi
+        if [ -d "h5p-questionnaire" ]; then cd h5p-questionnaire && npm install && npm run build; fi
+        if [ -d "h5p-drag-question" ]; then cd h5p-drag-question && npm install && npm run build; fi
+        if [ -d "h5p-interactive-video" ]; then cd h5p-interactive-video && npm install && npm run build; fi
+        if [ -d "h5p-open-ended-question" ]; then cd h5p-open-ended-question && npm install && npm run build; fi
+        if [ -d "h5p-simple-multiple-choice" ]; then cd h5p-simple-multiple-choice && npm install && npm run build; fi
```
Comment (kentis) [inline .github/workflows/build_and_test.yml:36]: hvorfor dette?

**Review by boyum — APPROVED** [review]

**Pull request merged**